modin/pandas/utils.py

```diff
@@ -80,6 +84,8 @@ def from_pandas(df):
     from modin.core.execution.dispatching.factories.dispatcher import FactoryDispatcher
     from .dataframe import DataFrame
 
+    if InitializeWithSmallQueryCompilers.get():
+        return DataFrame(query_compiler=SmallQueryCompiler(df))
```

Comment:
this feels completely off; I thought small query compiler should go the usual route for other QCs (in this case it should get picked by `FactoryDispatcher` according to user's settings)